### PR TITLE
Implement received bills ui

### PR DESCRIPTION
**Pull request by ArneSchulze**

### CODE REVIEW

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/adapter/BillsPagerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    private ReceivedBillsFragment receivedBillsFragment = new ReceivedBillsFragment();
+    private SentBillsFragment sentBillsFragment = new SentBillsFragment();
+
+    public BillsPagerAdapter(FragmentManager manager) {
+        super(manager);
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        if(position == 0) {
```
Comment (bugwelle) [inline app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/adapter/BillsPagerAdapter.java]: artistic style

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/utils/ServerCalls.java [hunk 2]
```diff
@@ -1120,6 +1122,115 @@ public void onDownloadProgress(long l, long l1, boolean b) {
...
+                        }
+                    }
+
+                    @Override
+                    public void onUploadProgress(long l, long l1, boolean b) {
+
+                    }
+
+                    @Override
+                    public void onDownloadProgress(long l, long l1, boolean b) {
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/utils/ServerCalls.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/UserDetail.java [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+import android.support.design.widget.FloatingActionButton;
+import android.support.design.widget.TextInputLayout;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.Toolbar;
+import android.transition.TransitionManager;
+import android.view.View;
+import android.view.animation.AnimationUtils;
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/UserDetail.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/WGPlanerApplication.java [hunk 4]
```diff
@@ -2,20 +2,34 @@
 
 import android.app.Application;
 
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.iid.FirebaseInstanceId;
+
+import de.ameyering.wgplaner.wgplaner.utils.Configuration;
 import de.ameyering.wgplaner.wgplaner.utils.DataProviderInterface;
+import de.ameyering.wgplaner.wgplaner.utils.ImageStore;
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/WGPlanerApplication.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/utils/ImageStore.java [hunk 5]
```diff
@@ -194,37 +140,29 @@ public boolean writeGroupMemberPicture(String uid, byte[] imageBytes, Context co
             }
         }
 
-        return success;
+        return;
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/utils/ImageStore.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/adapter/BillsContentAdapter.java [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+                if(bill.getSum() != null) {
+                    subHeader.setVisibility(View.VISIBLE);
+                    subHeader.setText(Double.valueOf(((double) bill.getSum()) / 100).toString());
+                } else {
+                    subHeader.setVisibility(View.GONE);
+                }
+
+                container.setOnClickListener(view -> callAllListeners(bill.getUid()));
+
+                switch (bill.getState()) {
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/adapter/BillsContentAdapter.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/WGPlanerApplication.java [hunk 7]
```diff
@@ -2,20 +2,34 @@
 
 import android.app.Application;
 
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.iid.FirebaseInstanceId;
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/WGPlanerApplication.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/BillDetailActivity.java [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+import android.transition.TransitionManager;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import java.text.DateFormat;
+import java.text.NumberFormat;
+import java.util.HashMap;
+import java.util.Locale;
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/BillDetailActivity.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/UserDetail.java [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.support.design.widget.FloatingActionButton;
+import android.support.design.widget.TextInputLayout;
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/UserDetail.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/UserDetail.java [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.Toolbar;
+import android.transition.TransitionManager;
+import android.view.View;
+import android.view.animation.AnimationUtils;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/UserDetail.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/EditEmailActivity.java [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+import de.ameyering.wgplaner.wgplaner.R;
+import de.ameyering.wgplaner.wgplaner.WGPlanerApplication;
+import de.ameyering.wgplaner.wgplaner.utils.DataProviderInterface;
+import de.ameyering.wgplaner.wgplaner.utils.OnAsyncCallListener;
+import io.swagger.client.ApiException;
+import io.swagger.client.model.User;
+
+public class EditEmailActivity extends AppCompatActivity {
+    private Toolbar toolbar;
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/EditEmailActivity.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/fragment/DatePickerFragment.java [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+
+public class DatePickerFragment extends DialogFragment implements DatePickerDialog.OnDateSetListener {
+
+    public interface OnDatePickedListener {
+
+        void onDatePicked(Date date);
+    }
+
+    private OnDatePickedListener listener = null;
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/fragment/DatePickerFragment.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/BillDetailActivity.java [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+import java.util.UUID;
+
+import de.ameyering.wgplaner.wgplaner.R;
+import de.ameyering.wgplaner.wgplaner.WGPlanerApplication;
+import de.ameyering.wgplaner.wgplaner.customview.CircularImageView;
+import de.ameyering.wgplaner.wgplaner.utils.DataProviderInterface;
+import de.ameyering.wgplaner.wgplaner.utils.ImageStore;
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/BillDetailActivity.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/fragment/BillsFragment.java [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+        viewPager.setCurrentItem(0);
+
+        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
+            @Override
+            public void onTabSelected(TabLayout.Tab tab) {
+                viewPager.setCurrentItem(tab.getPosition());
+            }
+
+            @Override
+            public void onTabUnselected(TabLayout.Tab tab) {
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/fragment/BillsFragment.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/UserDetail.java [hunk 15]
```diff
@@ -0,0 +1,260 @@
...
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.Toolbar;
+import android.transition.TransitionManager;
+import android.view.View;
+import android.view.animation.AnimationUtils;
+import android.widget.Button;
+import android.widget.EditText;
```
[the remark this answers is not in the document]
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/UserDetail.java]: Done

File: app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/adapter/BillsPagerAdapter.java (hunk 1 above)
Comment (ArneSchulze) [reply app/src/main/java/de/ameyering/wgplaner/wgplaner/section/home/adapter/BillsPagerAdapter.java]: Will do baby doll

**Pull request merged**